charts/backstage/templates/sonataflows.yaml

```
@@ -0,0 +1,133 @@
+          image: {{ .Values.orchestrator.sonataflowPlatform.dataIndexImage }}
+      {{- end }}
+    jobService:
+      enabled: true
+      persistence:
+        postgresql:
+          {{- if .Values.upstream.postgresql.enabled }}
+          secretRef:
+            name: {{ .Release.Name }}-postgresql-svcbind-postgres
+            userKey: username
+            passwordKey: password
+          serviceRef:
+            name: {{ .Release.Name }}-postgresql
+            namespace: {{ .Release.Namespace }}
+            databaseName: sonataflow
+          {{- else }}
+          secretRef:
+            name: {{ .Values.orchestrator.sonataflowPlatform.externalDBsecretRef}}
+            userKey: POSTGRES_USER
+            passwordKey: POSTGRES_PASSWORD
+          jdbcUrl: jdbc:postgresql://{{ .Values.upstream.backstage.appConfig.backend.database.connection.host}}:{{ .Values.upstream.backstage.appConfig.backend.database.connection.port}}/sonataflow?currentSchema=jobs-service
+          {{- end }}
+      {{- if .Values.orchestrator.sonataflowPlatform.jobServiceImage }}
+      podTemplate:
+        container:
+          image: {{ .Values.orchestrator.sonataflowPlatform.jobServiceImage }}
+      {{- end }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}-create-sonataflow-database
+  namespace: {{ .Release.Namespace }}
+spec:
+  template:
+    spec:
+      initContainers:
+        - name: wait-for-db
+          image: {{ .Values.orchestrator.sonataflowPlatform.initContainerImage }}
+          command:
+            - sh
+            - -c
+{{- if .Values.upstream.postgresql.enabled }}
+            - "until nc -z {{ .Release.Name }}-postgresql 5432; do echo 'Waiting for DB...'; sleep 2; done"
+{{- else }}
+            - "until nc -z {{ .Values.upstream.backstage.appConfig.backend.database.connection.host }} {{ .Values.upstream.backstage.appConfig.backend.database.connection.port }}; do echo 'Waiting for external DB...'; sleep 2; done"
+{{- end }}
+      containers:
+      - name: psql
+        image: {{ .Values.orchestrator.sonataflowPlatform.createDBJobImage }}
+        env:
+        - name: PGPASSWORD
+          valueFrom:
+{{- if .Values.upstream.postgresql.enabled }}
+            secretKeyRef:
+              name: {{ .Release.Name }}-postgresql-svcbind-postgres
+              key: password
+        command: [ "sh", "-c" ]
+        args:
+          - "psql -h {{ .Release.Name }}-postgresql -U postgres -c 'CREATE DATABASE sonataflow;'"
+{{- else }}
+            secretKeyRef:
+              name: {{ .Values.orchestrator.sonataflowPlatform.externalDBsecretRef}}
+              key: POSTGRES_PASSWORD
+        command: [ "sh", "-c" ]
+        args:
+          - "psql -h {{ .Values.upstream.backstage.appConfig.backend.database.connection.host }} -U {{ .Values.upstream.backstage.appConfig.backend.database.connection.user }} -d {{ .Values.orchestrator.sonataflowPlatform.externalDBName }} -c 'CREATE DATABASE sonataflow;'"
```

```suggestion
          - "psql -h {{ .Values.upstream.backstage.appConfig.backend.database.connection.host }} -U {{ .Values.upstream.backstage.appConfig.backend.database.connection.user }} -d {{ .Values.orchestrator.sonataflowPlatform.externalDBName }} -c 'CREATE DATABASE sonataflow;' || echo WARNING: Could not create database"
```

Smilar suggestion here.